management-interface.go

```diff
@@ -49,17 +52,36 @@ var tmpl *template.Template
 // This does some initialisation.  It parses our html templates up front and
 // finds the location where this executable was started.
 func init() {
+
+	deviceName = getDeviceName()
+
 	tmpl = template.New("")
 
 	for _, name := range templateBox.List() {
 		t := tmpl.New(name)
-		template.Must(t.Parse(templateBox.String(name)))
+		if name == "navbar.html" {
+			stringToParse := templateBox.String(name)
+			stringToParse = strings.Replace(stringToParse, "Cacophonator Management", deviceName, 1)
```

Comment:
It's a little odd to do this considering that we're already using templates!

It should be possible to pass the hostname into each ExecuteTemplate call with a little bit of extra work. It's OK to leave this for now though. If you leave it, please a comment about doing this better.

Reply:
Hmmm yes.  Ironic I know.

But if I don't do it there, then each call to ExecuteTemplate from each of the handlers would have to return that deviceName variable wouldn't it?

And some of the handlers return nil, and some of them return structures with data.  So I'd need to make them all return that device name inside a structure and use something like {{.DeviceName}} in the navbar template.

I couldn't really see how to just write that value into a header template once at startup.  I could do it in the init function, but ExecuteTemplate takes a writer as a parameter and I didn't know what to do with that.

Reply:
Menno, I did find a better way to do this.  I map a function to the template tree.  It is called from the navbar.html file when needed.  Hope that is better?

Reply:
Yep that's better. I was thinking you could have a function which generates the global variables to pass to each ExecuteTemplate call which then somehow get combined with any specific values for the template, but using a function is almost certainly simpler.